cognite_toolkit/_cdf_tk/resource_classes/transformations.py

```diff
@@ -0,0 +1,151 @@
+    version: str = Field(description="Version of the Data Model.")
+    destination_type: str = Field(description="External ID of the type(view) in the data model.")
+    destination_relationship_from_type: str | None = Field(
+        default=None, description="Property Identifier of the connection definition in destinationType."
+    )
+
+
+class ViewInfo(ToolkitResource):
+    space: str = Field(description="Space of the view.")
+    external_id: str = Field(description="External ID of the view.")
+    version: str = Field(description="Version of the view.")
+
+
+class EdgeType(ToolkitResource):
+    space: str = Field(description="Space of the type.")
+    external_id: str = Field(description="External ID of the type.")
+
+
+class DestinationSourceType(ToolkitResource):
+    type: Literal[
+        "assets",
+        "events",
+        "asset_hierarchy",
+        "datapoints",
+        "string_datapoints",
+        "timeseries",
+        "sequences",
+        "files",
+        "labels",
+        "relationships",
+        "data_sets",
+        "instances",
+        "nodes",
+        "edges",
+        "raw",
+        "sequence_rows",
+    ] = Field(description="The type of the destination resource.")
+
+
+class DataModelSource(DestinationSourceType):
+    data_model: DataModelInfo = Field(description="Target data model info.")
+    instance_space: str | None = Field(default=None, description="The space where the instances will be created.")
+
+
+class ViewDataSource(DestinationSourceType):
+    view: ViewInfo | None = Field(default=None, description="Target view info.")
+    edge_type: EdgeType | None = Field(default=None, description="Target type of the connection definition.")
+    instance_space: str | None = Field(
+        default=None, description="The space where the instances(nodes/edges) will be created."
+    )
+
+
+class RawDataSource(DestinationSourceType):
+    database: str = Field(description="The database name.")
+    table: str = Field(description="The table name.")
+
+
+class SequenceRowDataSource(DestinationSourceType):
+    external_id: str = Field(description="The externalId of sequence.")
+
+
+class AuthenticationClientIdSecret(ToolkitResource):
+    client_id: str = Field(description="Client Id.")
+    client_secret: str = Field(description="Client Secret.")
+    scopes: str | list[str] | None = Field(default=None, description="Scopes for the authentication.")
+    token_uri: str = Field(description="OAuth token url.")
+    cdf_project_name: str = Field(description="CDF project name.")
+    audience: str | None = Field(default=None, description="Audience for the authentication.")
+
+
+class OIDCCredential(ToolkitResource):
+    read: AuthenticationClientIdSecret | None = Field(
+        default=None, description="Source/Read authentication information."
+    )
+    write: AuthenticationClientIdSecret | None = Field(
+        default=None, description="Destination/Write authentication information."
+    )
```

Comment:
There is one option where the user gives only `clientId/clientSecret` and one option where they also specify `cdf_project_name`, `token_uri`, `scopes` and optionally `audience`.